subcommand/health-sync/checks.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package healthsync
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/config"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-multierror"
+)
+
+// fetchHealthChecks fetches the Consul health checks for both the service
+// and proxy registrations
+func (c *Command) fetchHealthChecks(consulClient *api.Client, taskMeta awsutil.ECSTaskMeta) (map[string]*api.HealthCheck, error) {
+	serviceName := c.constructServiceName(taskMeta.Family)
+	serviceID := makeServiceID(serviceName, taskMeta.TaskID())
+	proxySvcID, proxySvcName := makeProxySvcIDAndName(serviceID, serviceName)
+
+	healthCheckMap := make(map[string]*api.HealthCheck)
```

Comment:
generally we are advised to use `var healthCheckMap map[string]*api.HealthCheck`

Reply:
Why is that? The zero valued initialization for a map type is not an empty map but `nil`. The behaviour is in contrast to string and bool types. We will end up with `nil` panics if we follow zero valued initialization here. Also it is safe to assume that the `healthCheckMap` will contain atleast one entry by the end of this function and so initialization doesn't won't be wasted.